presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2054,6 +2055,112 @@ public void testGrantRevokeNonExistingRoles()
         assertUpdate("DROP ROLE grant_revoke_role_existing_1");
     }
 
+    @Test
+    public void testSetRole()
+            throws Exception
+    {
+        assertUpdate("CREATE ROLE set_role_1");
+        assertUpdate("CREATE ROLE set_role_2");
+        assertUpdate("CREATE ROLE set_role_3");
+        assertUpdate("CREATE ROLE set_role_4");
+        assertUpdate("GRANT set_role_1 TO USER set_user_1");
+        assertUpdate("GRANT set_role_2 TO ROLE set_role_1");
+        assertUpdate("GRANT set_role_3 TO ROLE set_role_2");
+
+        Session unsetRole = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .build();
+        Session setRoleAll = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.ALL, Optional.empty()))
+                .build();
+        Session setRoleNone = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.NONE, Optional.empty()))
+                .build();
+        Session setRole1 = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("set_role_1")))
+                .build();
+        Session setRole2 = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("set_role_2")))
+                .build();
+        Session setRole3 = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("set_role_3")))
+                .build();
+        Session setRole4 = Session.builder(queryRunner.getDefaultSession())
+                .setIdentity(new Identity("set_user_1", Optional.empty()))
+                .setRole("hive", new SelectedRole(SelectedRole.Type.ROLE, Optional.of("set_role_4")))
+                .build();
+
+        MaterializedResult actual = queryRunner.execute(unsetRole, "SELECT * FROM hive.information_schema.applicable_roles");
+        MaterializedResult expected = MaterializedResult.resultBuilder(unsetRole, createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType(), createUnboundedVarcharType())
+                .row("set_user_1", "USER", "admin", "NO")
```

Comment:
I changed that already. Now only `hive`, `hdfs` and `admin` users are admins when using file based hive metastore.